discord/abc.py

```diff
@@ -878,6 +888,12 @@ async def send(self, content=None, *, tts=False, embed=None, file=None,
         else:
             allowed_mentions = state.allowed_mentions and state.allowed_mentions.to_dict()
 
+        if reference is not None:
+            if isinstance(reference, _MessageType):
+                reference = (reference if isinstance(reference, MessageReference) else reference.to_reference()).to_dict()
```

Comment:
this inline if statement isn't exactly readable; something like this could suffice it.
```suggestion
                if not isinstance(reference, MessageReference):
                    reference = reference.to_reference()
                reference = reference.to_dict()
```